### DISCUSSION TITLE
Print recovery after a mcu communication lost

### DISCUSSION THREAD

**Discussion by Robourek** [mainsail-crew/mainsail#1854] [General]
Hello,
is there any way to save the line of the gcode in the case of lost communication with mcu for easier print recover? thank you in andvance.

**Comment by meteyou** [comment]
In the Klipper-Log, there is already the Byte position from the Gcode-File. There are also some unofficial Klipper plugins for that, but i don't have a link to it right now...

But to be honest. This is more a Klipper question, then a Mainsail question. Mainsail is only a webinterface. Nothing more. So i think you will get a better answer, if you ask in a Klipper specific channel.
